src/vos/vos_io.c

```diff
@@ -2447,7 +2447,8 @@ vos_update_end(daos_handle_t ioh, uint32_t pm_ver, daos_key_t *dkey, int err,
 	tx_started = true;
 
 	/* Commit the CoS DTXs via the IO PMDK transaction. */
-	if (dtx_is_valid_handle(dth) && dth->dth_dti_cos_count > 0 && !dth->dth_cos_done) {
+	if (dtx_is_valid_handle(dth) && dth->dth_dti_cos_count > 0 && !dth->dth_cos_done &&
+	    !vos_pool_is_evictable(vos_cont2pool(ioc->ic_cont))) {
```

Comment:
Disable DTX commit-on-share (CoS) if pool is evictable? Some potential effect for that:
1. Some unexpected DER_INPROGRESS failure on non-leader under the case of leader does successfully.
2. Decrease the efficiency of DTX commit.

Reply:
I'm not sure if I follow the 1 item, let's discuss it offline.